libs/vkd3d/command.c

```diff
@@ -4736,6 +4744,38 @@ static void STDMETHODCALLTYPE d3d12_command_list_OMSetStencilRef(d3d12_command_l
     dyn_state->dirty_flags |= VKD3D_DYNAMIC_STATE_STENCIL_REFERENCE;
 }
 
+#ifdef VKD3D_ENABLE_RENDERDOC
+static void d3d12_command_list_check_renderdoc_capture(struct d3d12_command_list *list,
+        struct d3d12_pipeline_state *state)
+{
+    unsigned int i;
+
+    if (vkd3d_renderdoc_active() && state)
+    {
+        if (state->vk_bind_point == VK_PIPELINE_BIND_POINT_COMPUTE)
+        {
+            if (vkd3d_renderdoc_should_capture_shader_hash(state->compute.meta.hash))
+            {
+                WARN("Triggering RenderDoc capture for this command list.\n");
+                list->debug_capture = true;
+            }
+        }
+        else if (state->vk_bind_point == VK_PIPELINE_BIND_POINT_GRAPHICS)
+        {
+            for (i = 0; i < state->graphics.stage_count; i++)
+            {
+                if (vkd3d_renderdoc_should_capture_shader_hash(state->graphics.stage_meta[i].hash))
+                {
+                    WARN("Triggering RenderDoc capture for this command list.\n");
+                    list->debug_capture = true;
+                    break;
+                }
+            }
+        }
+    }
+}
+#endif
```

Comment:
Not a huge fan of having entire renderdoc-specific functions in common code; can these be moved to renderdoc.c instead?

Reply:
Fixed.